doc/releases/release-notes-3.7.rst

```diff
@@ -26,6 +26,68 @@ Mbed TLS was updated to version 3.6.2 (from 3.6.0). The release notes can be fou
 Mbed TLS 3.6 is an LTS release that will be supported
 with security and bug fixes until at least March 2027.
 
+MCUboot
```

Comment:
this is too verbose of a release note in a module to be included in the zephyr release notes, dont you have something to link to similar to what we do tfm and mbedtls?

Reply:
the issue i see here is that all of those changes might be seen as changes to zephyr and not mcuboot itself, which might be confusing.

Reply:
MCUboot has release notes but they're only generated when a release is made, and these are cherry-picked commits so if a release was made, it wouldn't be correct as it's be for mcuboot's `main` not the zephyr-only branch with cherry picked commits

Reply:
Can just change to link to this PR instead?

Reply:
> You can still make bug fix releases on the branch, no?

It's a zephyr branch of MCUboot, it does not exist upstream. https://github.com/zephyrproject-rtos/mcuboot/tree/v3.7-branch is where it gets cherry picked to

> Either remove them from this PR

Will do